tests/org.eclipse.papyrus.uml.interaction.model.tests/src/org/eclipse/papyrus/uml/interaction/model/tests/creation/CreateMessageTestB.java

```diff
@@ -0,0 +1,185 @@
+		message5Top = interaction().getMessages().get(3).getTop().getAsInt();
+	}
+
+	private DiagramHelper diagramHelper() {
+		return LogicalModelPlugin.getInstance().getDiagramHelper(model.getEditingDomain());
+	}
+
+	private LayoutHelper layoutHelper() {
+		return LogicalModelPlugin.getInstance().getLayoutHelper(model.getEditingDomain());
+	}
+
+	private int getLifelineBodyTop(MLifeline lifeline) {
+		View shape = lifeline.getDiagramView().orElse(null);
+		return layoutHelper()//
+				.getTop(diagramHelper().getLifelineBodyShape(shape));
+	}
+
+	private MInteraction interaction() {
+		if (interaction == null) {
+			interaction = MInteraction.getInstance(model.getInteraction(), model.getSequenceDiagram().get());
+		}
+		return interaction;
+	}
+
+	private void execute(Command remove) {
+		if (!remove.canExecute()) {
+			Assert.fail("Command not executable"); //$NON-NLS-1$
+		}
+		remove.execute();
+		/* force reinit after change */
+		interaction = null;
+	}
+
+	@Test
+	public void creationMessage1() {
+		/* setup */
+		MLifeline lifeline1 = interaction().getLifelines().get(0);
+		MLifeline lifeline2 = interaction().getLifelines().get(1);
+		CreationCommand<Message> command = lifeline1.insertMessageAfter(lifeline1, 10, lifeline2,
+				MessageSort.CREATE_MESSAGE_LITERAL, null);
+
+		/* act */
+		execute(command);
+
+		/* assert: Create Message created */
+		/* assert: Lifeline1, Lifeline4 should not move */
+		/* assert: Lifeline2 moved down */
+		/* assert: other elements nudged down */
+		assertEquals(6, interaction().getMessages().size());
+		assertEquals(MessageSort.CREATE_MESSAGE_LITERAL,
+				interaction().getMessages().get(5).getElement().getMessageSort());
+		assertEquals(getLifelineBodyTop(interaction().getLifelines().get(0)) + 10,
+				interaction().getMessages().get(5).getTop().getAsInt());
+
+		assertEquals(lifeline1Top, interaction().getLifelines().get(0).getTop().getAsInt());
+		assertEquals(lifeline4Top, interaction().getLifelines().get(3).getTop().getAsInt());
+
+		int nudgedLifeline2Top = interaction().getMessages().get(5).getTop().getAsInt()
+				- (lifeline2Header / 2);
+		assertEquals(nudgedLifeline2Top, interaction().getLifelines().get(1).getTop().getAsInt());
+
+		int delta = nudgedLifeline2Top - lifeline2Top;
+		assertEquals(lifeline3Top + delta, interaction().getLifelines().get(2).getTop().getAsInt());
+		assertEquals(message1Top + delta, interaction().getMessages().get(0).getTop().getAsInt());
+		assertEquals(message2Top + delta, interaction().getMessages().get(4).getTop().getAsInt());
+		assertEquals(message3Top + delta, interaction().getMessages().get(1).getTop().getAsInt());
+		assertEquals(message4Top + delta, interaction().getMessages().get(2).getTop().getAsInt());
+		assertEquals(message5Top + delta, interaction().getMessages().get(3).getTop().getAsInt());
+		assertEquals(message1Top + delta, interaction().getMessages().get(0).getBottom().getAsInt());
+		// TODO creation bottom
```

Comment:
Please avoid todos in the code. Either we should open a new issue for this matter, or we should fix it within this PR, if it is reasonable.